src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -455,28 +503,39 @@ void activity_classifier::init_train(
     std::string session_id_column_name, variant_type validation_data,
     std::map<std::string, flexible_type> opts)
 {
-
+  gl_sframe train_data;
+  gl_sframe val_data;
   // Begin printing progress.
   // TODO: Make progress printing optional.
   if (variant_is<gl_sframe>(validation_data)) {
-    gl_sframe validation_sf = variant_get_value<gl_sframe>(validation_data);
-    if (!validation_sf.empty()) {
-      training_table_printer_.reset(
-          new table_printer({{"Iteration", 12},
-                             {"Train Accuracy", 12},
-                             {"Train Loss", 12},
-                             {"Validation Accuracy", 12},
-                             {"Validation Loss", 12},
-                             {"Elapsed Time", 12}}));
+    train_data = data;
+    val_data = variant_get_value<gl_sframe>(validation_data);
+    if (!val_data.empty()) {
+      init_table(true);
     } else {
       log_and_throw("Input SFrame either has no rows or no columns. A "
                     "non-empty SFrame is required");
     }
+  }
+  else if ((variant_is<flex_string>(validation_data)) && (variant_get_value<flex_string>(validation_data)=="auto")) {
+    gl_sarray unique_session = data[session_id_column_name].unique();
+    if (unique_session.size() > MIN_NUM_OF_SESSIONS) {
+      std::tie(train_data, val_data) = random_split_by_session(data, session_id_column_name, 0.9, 1);
```

Comment:
Let's pull the `0.9` constant into a named constant at the top of the file. We could also use a TODO to expose the seed parameter publicly